iroh/src/client/blobs.rs

```diff
@@ -368,6 +390,263 @@ where
     }
 }
 
+/// A scope in which blobs can be added.
+#[derive(derive_more::Debug)]
+struct BatchInner<C: ServiceConnection<RpcService>> {
+    /// The id of the scope.
+    scope: u64,
+    /// The rpc client.
+    rpc: RpcClient<RpcService, C>,
+    /// The stream to send drop
+    #[debug(skip)]
+    updates: Mutex<UpdateSink<RpcService, C, BatchUpdate>>,
+}
+
+/// A batch for write operations.
+///
+/// This serves mostly as a scope for temporary tags.
+///
+/// It is not a transaction, so things in a batch are not atomic. Also, there is
+/// no isolation between batches.
+#[derive(derive_more::Debug)]
+pub struct Batch<C: ServiceConnection<RpcService>>(Arc<BatchInner<C>>);
+
+impl<C: ServiceConnection<RpcService>> TagDrop for BatchInner<C> {
+    fn on_drop(&self, content: &HashAndFormat) {
+        let mut updates = self.updates.lock().unwrap();
+        updates.send(BatchUpdate::Drop(*content)).now_or_never();
+    }
+}
+
+impl<C: ServiceConnection<RpcService>> Batch<C> {
+    /// Write a blob by passing bytes.
+    pub async fn add_bytes(&self, bytes: impl Into<Bytes>, format: BlobFormat) -> Result<TempTag> {
+        let input = futures_lite::stream::once(Ok(bytes.into()));
+        self.add_stream(input, format).await
+    }
+
+    /// Import a blob from a filesystem path.
+    ///
+    /// `path` should be an absolute path valid for the file system on which
+    /// the node runs, which refers to a file.
+    ///
+    /// If `import_mode` is TryReference, Iroh will assume that the data will not
```

Comment:
maybe write "If you set`ImporteMode::TryReference` ... ", makes it a bit easier (for me at least) to understand that you mean the enum variant